packages/hardhat-core/src/internal/core/jsonrpc/types/input/hardhat-network.ts

```diff
@@ -0,0 +1,53 @@
+import * as t from "io-ts";
+
+import { optional } from "../../../../util/io-ts";
+import { rpcUnsignedInteger } from "../base-types";
+
+export const rpcForkConfig = optional(
```

Comment:
I put this types here, despite they being hardhat network specific. IDK what should be done instead 🤷‍♂️ We'll probably improve this once we plan how to extract Hardhat Network outside of core.